Move package section, and add tutorial & index page (#2035)
umbraco/UmbracoDocs: Extending/Packages/Creating-a-Package/index.md

![Buttons to select for creating a package in the backoffice](images/creating-package-menu.png)

On the `Create package` page there are 4 sections with different info to fill out. Some of the info is mandatory, most of it is not. First of all you should give the package a name at the top - we will call our dashboard the same as in the [tutorial](../Creating-a-Custom-Dashboard/index.md): `Custom Welcome Dashboard`.

### The Package Properties section

This section contains mostly meta data about the package and the creator. We will fill in the following things:

| Property | Value | Note |
| -------- | ----- | ---- |
| Url | https://umbraco.com | This url will be shown as the "packages url" when others install it. Will mostly be a Github repo or similar |
| Version | 1.0.0 | This is automatically set to 1.0.0 but can be changed or increased manually if needed |
| Icon Url | _Blank_ | Not a mandatory value, but will appear as the package icon different places in the backoffice if set |

> | Umbraco version | 8.2.0 | It will automatically select the version your Umbraco is currently on. This is then set as the compatible version for the package |

Reviewer: ```suggestion
| Umbraco version | 8.2.0 | It will automatically select the Umbraco version you are currently using. This is then set as the compatible version for the package |
```

### The Package Content section

This section is used to determine which things the package should contain. We will fill in the following things:

| Property | Value | Note |
| -------- | ----- | ---- |
| Content | _Empty_ | Here you can include content - fx if you want to create a starter kit. Not relevant for this package though |
| Document Types | _Empty_ | Similar to the Content picker above. Important to note that if you include content you will need to also pick all its dependencies in this and the next steps for it to be packaged together! |
| Templates | _Empty_ | See `Document Types` above |
| Stylesheets | _Empty_ | This will come from the ~/css folder. If you have stylesheets you want to include from other locations you can do so at a later step |
| Macros | _Empty_ | See `Document Types` above |
| Languages | _Empty_ | See `Document Types` above - all text is hardcoded or within our own lang folder in this package, so this is not needed |
| Dictionary | _Empty_ | See `Document Types` above |
| Data Types | _Empty_ | See `Document Types` above |

### The Package Files section

> In this section you will be able to select all of your own custom files. Under the `Path to file` option. since everything in our Dashboard is from the same folder in App_Plugins, we can select the folder and it will include all items inside that folder:

Reviewer: ```suggestion
In this section you will be able to select all of your own custom files. We will start with the `Path to file` option.

Since everything in our Dashboard is from the same folder in `App_Plugins`, we can select the folder and it will include all items inside that folder:
```

![Selecting the files you want in your package](images/select-files-for-package.png)

> We will leave the `Package options view` selector empty, but in case you were wondering you can select an html file here that will show up as package options. Here is an example from the package [Articulate](https://github.com/Shazwazza/Articulate/tree/master/src/Articulate.Web/App_Plugins/Articulate/BackOffice/PackageOptions):

Reviewer: ```suggestion
We will leave the `Package options view` selector empty, but in case you were wondering you can select an HTML file here that will show up as package options. Here is an example from the package [Articulate](https://github.com/Shazwazza/Articulate/tree/master/src/Articulate.Web/App_Plugins/Articulate/BackOffice/PackageOptions):
```

![Gif of Articulates use of Package options](images/package-options.gif)

### The Package Actions section

> Here you can add package actions. You can read more about the default package actions [here](../../Reference/Packaging/index.md). You can also create your own package action and reference it here, to do so you will need to implement the IPackageAction interface, you can see an example [here](https://github.com/Shazwazza/Articulate/blob/master/src/Articulate/Packaging/ArticulateInstallPackageAction.cs).

Reviewer: I've tried to rewrite this, to not use `here` for the links.

```suggestion
Here you can add package actions. There are a number of [default package actions](../../Reference/Packaging/index.md) and you can also create your own package action and reference it here, to do so you will need to implement the IPackageAction interface, as in the [Articulate Package](https://github.com/Shazwazza/Articulate/blob/master/src/Articulate/Packaging/ArticulateInstallPackageAction.cs).
```

> Finally after filling out all the info we can either `Save` or `Download` the package. We will download it, in order to take a closer look at what it contains in the zip.

Reviewer: ```suggestion
Finally after filling out all the info we can either `Save` or `Download` the package. We will download it, in order to take a closer look at what it contains in the generated zip file.
```

Reviewer: I have one option: `Create` ? 🤔 
So, shouldn't that step be specified before `Save` and `Download` ?

      <orgPath>~/App_Plugins/CustomWelcomeDashboard</orgPath>
      <orgName>WelcomeDashboard.html</orgName>
    </file>
    <file>
      <guid>en-US.xml</guid>
      <orgPath>~/App_Plugins/CustomWelcomeDashboard/lang</orgPath>
      <orgName>en-US.xml</orgName>
    </file>
  </files>
  <info>
    <package>
      <name>Custom Welcome Dashboard</name>
      <version>1.0.0</version>
      <iconUrl></iconUrl>
      <license url="http://opensource.org/licenses/MIT">MIT License</license>
      <url>https://github.com/umbraco/customwelcomedashboard</url>
      <requirements type="Strict">
        <major>8</major>
        <minor>2</minor>
        <patch>0</patch>
      </requirements>
    </package>
    <author>
      <name>Jesper Mayntzhusen</name>
      <website>https://jesper.com</website>
    </author>
    <contributors></contributors>
    <readme><![CDATA[This will add a dashboard to your content section. 

The dashboard will show each user the most recent nodes they have saved.]]></readme>
  </info>
  <DocumentTypes />
  <Templates />
  <Stylesheets />
  <Macros />
  <DictionaryItems />
  <Languages />
  <DataTypes />
  <Actions />
</umbPackage>
```

> You will notice that each of the fields we created is inside this XML file. And all files have an `orgPath` attached to it that it will try to move the file to when installing it - so don't worry if you organized your package in folders and they are not so in the zip.

Reviewer: ```suggestion
You will notice that each of the fields we created is inside this XML file. All files have an `orgPath` attached to it - this is where it will try to move the file to when installing it. So, don't worry if you organized your package in folders and they are not so in the zip.
```
